### PR TITLE
API version 2.1.0 See https://docs.patch.io/#/changelog for changes

### PR DESCRIPTION
**Pull request by venalen**
### What

API version 2.1.0. See https://docs.patch.io/#/changelog for changes

### Why

Version 2.1.0 adds support for 
* `vintage_start_year` and `vintage_end_year` for orders
* `vintage_start_year` and `vintage_end_year` for  order line items
* `vintage_start_year` and `vintage_end_year` for  inventory
*  `carrier_scac` field to `patch.estimates.create_road_shipping_estimate`

### SDK Release Checklist

- [x] Have you added an integration test for the changes?
- [x] Have you built the library locally and made queries against it successfully?
- [x] Did you update the changelog?
- [x] Did you bump the package version?
- [ ] ~If endpoints were removed, did you manually remove the corresponding files? (this should be rare)~
- [ ] ~For breaking changes, did you plan for the release of the new SDK versions and deploy the API to production?~

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v2
         with:
-          python-version: '3.6'
+          python-version: '3.10'
```
[the remark this answers is not in the document]
Comment (pcothenet) [reply .github/workflows/test.yml:26]: Agree. But I'm not sure we should change the required version for the package: https://github.com/patch-technology/client-code-generation/pull/70#pullrequestreview-1359412217

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ pip install patch-api
 
 ### Requirements
 
-- Python 3.6.1
+- Python 3.10
```
Comment (pcothenet) [inline README.md:35]: As discussed on https://github.com/patch-technology/client-code-generation/pull/70#pullrequestreview-1359412217, are we sure we want to deprecate Python < 3.10. Seems a bit aggressive? (but I'm not up-to-date on the Python ecosystem)

File: .github/workflows/test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (venalen) [reply .github/workflows/test.yml:26]: Oh yeah, that makes sense. I'll also poke around to make sure that we need this at all; the `publish.yml` file still has a version of 3.6, so I'm wondering if this needs any change at all

**Comment by venalen** [comment]
@pcothenet thanks for the nudge to look into the python issues more deeply! It looks like pinning the ubuntu version remedies the situation and avoids an unnecessary python upgrade. It seems like there's an underlying issue with the python tarballs for `ubuntu-latest` which will hopefully be addressed in the near future, but for now we can use another os version and it should have impact on the SDK/python itself. WDYT?

**Comment by pcothenet** [comment]
> @pcothenet thanks for the nudge to look into the python issues more deeply! It looks like pinning the ubuntu version remedies the situation and avoids an unnecessary python upgrade. It seems like there's an underlying issue with the python tarballs for `ubuntu-latest` which will hopefully be addressed in the near future, but for now we can use another os version and it should have impact on the SDK/python itself. WDYT?

Sounds good to me.

I think you could also keep the test running on 3.10, but not force that as a min version for the SDK (ultimately, we should have a test matrix of multiple Python versions for this SDK)

**Comment by venalen** [comment]
Ah yeah a test matrix would be ideal and hopefully is in our near future. For now I was thinking that it would be best to have the tests run on the same version that the package is published to maintain consistency!

File: test/test_estimates_api.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def test_create_and_retrieve_shipping_estimate(self):
             create_order=False,
         )
         self.assertEqual(estimate.data.order, None)
-        self.assertEqual(estimate.data.type, "shipping")
+        self.assertEqual(estimate.data.type, "ecommerce")
```
Comment (venalen) [inline test/test_estimates_api.py:92]: We're now re-routing to the ecommerce endpoint: https://github.com/patch-technology/patch/pull/5588

**Pull request merged**